imports/server/accounts.ts

```diff
@@ -55,32 +51,115 @@ const summaryFromLoginInfo = function (info: LoginInfo) {
 
 Accounts.registerLoginHandler(async (options: LoginOptions) => {
   // Only handle requests that include our hook's custom flag.
-  if (!options.isGoogleJrLogin) {
+  if (!options.isJrLogin) {
     return undefined;
   }
 
   check(options, {
-    isGoogleJrLogin: true,
-    key: Match.Optional(String),
-    secret: Match.Optional(String),
+    isJrLogin: true,
+    googleCredentials: Match.Optional({
+      key: String,
+      secret: String,
+    }),
+    allowAutoProvision: Match.Optional({
+      huntInvitationCode: String,
+    }),
   });
 
-  if (!options.key || !options.secret) {
+  // If an invitation code is provided, create a new user if needed.
+  // We don't actually add the user to the hunt here - that should be handled once the user is
+  // signed in and redirected to the /join URL they originally tried to access.
+  if (options.allowAutoProvision) {
+    if (!options.googleCredentials) {
+      throw new Meteor.Error(
+        400,
+        "Autoprovisioning login request is missing Google credentials",
+      );
+    }
+
+    const invitation = await InvitationCodes.findOneAsync({
+      code: options.allowAutoProvision.huntInvitationCode,
+    });
+    if (!invitation) {
+      throw new Meteor.Error(404, "Invalid invitation code");
+    }
+
+    const hunt = await Hunts.findOneAsync({
+      _id: invitation.hunt,
+    });
+    if (!hunt) {
+      throw new Meteor.Error(404, "Hunt does not exist for invitation");
+    }
+
+    // Google credentials were provided - obtain the email from the associated account,
+    // create a passwordless account with that email, and link it to the Google account.
+    // Also set the initial displayName for the user to their Google account's name.
+    const credential = await Google.retrieveCredential(
+      options.googleCredentials.key,
+      options.googleCredentials.secret,
+    );
+    const { email, id, name, picture } = credential.serviceData;
+    const users = await MeteorUsers.find(
+      { googleAccountId: id },
+      { limit: 2 },
+    ).fetchAsync();
+    switch (users.length) {
+      case 0: {
+        // Autoprovision a new user
+        const userId = await Accounts.createUserAsync({
+          email,
+        });
+        await MeteorUsers.updateAsync(userId, {
+          $set: {
+            googleAccount: email,
+            googleAccountId: id,
+            googleProfilePicture: picture,
+            displayName: name,
```

Comment:
Is this guaranteed to be populated?

I'm guessing not, and that if it isn't, we have the fallback of the site notifying you to set one while still not having things fall apart in the meantime, but just want to make sure it'd be handled reasonably gracefully.